Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -32,8 +35,15 @@ public object Deserialize(IIonReader reader)
                 FieldInfo field;
                 if (property != null)
                 {
+                    if (IsReadOnlyProperty(property))
```

Comment:
`IgnoreReadOnlyProperties` only applies to Serialization.

Reply:
We still need this check regardless of the `IgnoreReadOnlyProperties` option because without it `property.SetValue()` will throw an exception on the read-only property.

Because of this check we still need to handle the scenario where we are deserializing read-only properties (when the `IgnoreReadOnlyProperties` option is false). Instead of calling `property.SetValue()`, we deserialize the corresponding backing field instead.